### PR TITLE
(Doc+) Allocation Explain Examples: THROTTLED, MAX_RETRY

### PR DESCRIPTION
**Pull request by stefnestor**
👋🏽 howdy, team! 

Adds [Allocation Explain examples](https://www.elastic.co/guide/en/elasticsearch/reference/master/cluster-allocation-explain.html#cluster-allocation-explain-api-examples) for `THROTTLED` and `MAX_RETRY`. Also formats sub TOC so that we can after link code message to those docs.

### CODE REVIEW

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 1]
```diff
@@ -158,6 +159,56 @@ node.
 <5> The decider which led to the `no` decision for the node.
 <6> An explanation as to why the decider returned a `no` decision, with a helpful hint pointing to the setting that led to the decision. In this example, a newly created index has <<indices-get-settings,an index setting>> that requires that it only be allocated to a node named `nonexistent_node`, which does not exist, so the index is unable to allocate.
 
+====== Max Retries
```
Comment (shainaraskas) [inline docs/reference/cluster/allocation-explain.asciidoc:162]:
```suggestion
====== Maximum number of retries exceeded
```

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 2]
```diff
@@ -158,6 +159,56 @@ node.
...
+
+The following response contains an allocation explanation for an unassigned
+primary shard that has reached max allocation retry attempts.
```
Comment (shainaraskas) [inline docs/reference/cluster/allocation-explain.asciidoc:165]:
```suggestion
The following response contains an allocation explanation for an unassigned
primary shard that has reached the maximum number of allocation retry attempts. 
```

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 3]
```diff
@@ -158,6 +159,56 @@ node.
...
+        }
+      ]
+    }
+  ]
+}
+----
+// NOTCONSOLE
+
+If decider message indicates a transient allocation issue, use 
+<<cluster-reroute,Cluster Reroute>> to retry allocation.
```
Comment (shainaraskas) [inline docs/reference/cluster/allocation-explain.asciidoc:208]:
```suggestion
<<cluster-reroute,the cluster reroute API>> to retry allocation. 
```

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 4]
```diff
@@ -241,8 +294,52 @@ unassigned due to <<delayed-allocation,delayed allocation>>.
...
+          "explanation": "reached the limit of incoming shard recoveries [2], cluster setting [cluster.routing.allocation.node_concurrent_incoming_recoveries=2] (can also be set via [cluster.routing.allocation.node_concurrent_recoveries])"
+        }
+      ]
+    }
+  ]
+}
+----
+// NOTCONSOLE
+
+This is a transient message which may appear while a large amount of shards are allocating.
```
Comment (shainaraskas) [inline docs/reference/cluster/allocation-explain.asciidoc:337]:
```suggestion
This is a transient message that might appear when a large amount of shards are allocating. 
```

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 5]
```diff
@@ -338,7 +437,7 @@ cluster balance.
 ===== No arguments
 
 If you call the API with no arguments, {es} retrieves an allocation explanation
-for an arbitrary unassigned primary or replica shard.
+for an arbitrary unassigned primary or replica shard, with preference to primary.
```
Comment (shainaraskas) [inline docs/reference/cluster/allocation-explain.asciidoc:440]:
```suggestion
for an arbitrary unassigned primary or replica shard, returning any unassigned primary shards first. 
```

**Review by shainaraskas — APPROVED** [review]
thanks for these additions! provided some small language feedback and also added a couple of missing labels 👍

**Review by shainaraskas — APPROVED** [review]

**Review by DiannaHohensee — APPROVED** [review]
LGTM 👍

**Pull request merged**